File: Patterns/Facade Pattern/CSharp/HomeTheater/CdPlayer.cs
using System;

namespace HomeTheater
{
    public class CdPlayer
    {
        private string description;
        private int currentTrack;
        private string? title;

        public CdPlayer(string description)
        {
            this.description = description;
        }

        public void On() => Console.WriteLine($"{description} on");

        public void Off() => Console.WriteLine($"{description} off");

        public void Eject()
        {
            title = null;
            Console.WriteLine($"{description} eject");
        }

        public void Play(string title)
        {
            this.title = title;
            currentTrack = 0;
            Console.WriteLine($"{description} playing \"{title}\"");
        }

        public void Play(int track)
        {
            if (title == null)
            {
                Console.WriteLine($"{description} can't play track {currentTrack}, no cd inserted");
            }
            else
            {
                currentTrack = track;
                Console.WriteLine($"{description} playing track {currentTrack}");
            }
        }

        public void Stop() {
            currentTrack = 0;
            Console.WriteLine($"{description} stopped");
        }

        public void Pause() => Console.WriteLine($"{description} pause \"{title}\"");

        public string Description { get => description; private set => description = value; }
    }
}

File: Patterns/Facade Pattern/Cpp/HomeTheater/CdPlayer.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;

class CdPlayer {
private:
    string description;
    int currentTrack;
    string title;

public:
    CdPlayer(string description) : description{description} {}
    
    void on() {
        cout << description << " on\n";
    }

    void off() {
        cout << description << " off\n";
    }

    void eject() {
        title = "";
        cout << description << " eject\n";
    }

    void play(string title) {
        this->title = title;
        currentTrack = 0;
        cout << description << " playing \"" << title << "\"\n";
    }

    void play(int track) {
        if (title.empty()) {
            cout << description << " can't play track " << currentTrack << ", no cd inserted\n";
        }
        else {
            currentTrack = track;
            cout << description << " playing track " << currentTrack << "\n";
        }
    }

    void stop() {
        currentTrack = 0;
        cout << description << " stopped\n";
    }

    void pause() {
        cout << description << " paused \"" << title << "\"\n";
    }

    string toString() {
        return description;
    }
};